### PR TITLE
fix ban-unused-ignore and ban-unknown-rule-code

### LINKED ISSUE #502: "ban-unknown-rule-code" should distinguish between known and unknown rules
Configuring linter to run only a single rule will cause a lot of `ban-unknown-rule-code` errors
due to other rules not being run. This is wrong - `ban-unknown-rule-code` should know
all available rules codes even if certain rule is not being run. 

Instead, `ban-unused-ignore` should be raised in case of running with only a single rule configured.

Ref https://github.com/denoland/deno_lint/pull/499#discussion_r516777180

### PR DESCRIPTION
**Pull request by magurotuna**
Fixes #502

### CODE REVIEW

File: src/linter.rs [hunk 1]
```diff
@@ -236,10 +236,17 @@ impl Linter {
...
 
-    let rule_codes = rules
+    #[allow(clippy::borrowed_box)]
+    fn to_code(rule: &Box<dyn LintRule>) -> String {
+      rule.code().to_string()
+    }
+
+    let executed_rule_codes =
+      rules.iter().map(to_code).collect::<HashSet<String>>();
+    let available_rule_codes = get_recommended_rules()
```
Comment (bartlomieju) [inline src/linter.rs:246]: This should probably be `get_all_rules()`

File: src/linter.rs [hunk 2]
```diff
@@ -236,10 +236,17 @@ impl Linter {
     let ignore_directives = context.ignore_directives.clone();
     let diagnostics = &context.diagnostics;
 
-    let rule_codes = rules
+    #[allow(clippy::borrowed_box)]
+    fn to_code(rule: &Box<dyn LintRule>) -> String {
+      rule.code().to_string()
+    }
```
Comment (bartlomieju) [inline src/linter.rs:242]: IMO superfluous, please remove

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: src/linter.rs (hunk 2 above)
Comment (magurotuna) [reply src/linter.rs:242]: Alright, I'll remove soon

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**